internal/jobcontainers/jobcontainer.go

```diff
@@ -174,14 +174,18 @@ func (c *JobContainer) CreateProcess(ctx context.Context, config interface{}) (_
 		return nil, errors.New("console emulation not supported for job containers")
 	}
 
-	absPath, commandLine, err := getApplicationName(conf.CommandLine, c.sandboxMount, os.Getenv("PATH"))
+	commandLine := c.replaceWithMountPoint(conf.CommandLine)
+
+	workDir := c.sandboxMount
+	if conf.WorkingDirectory != "" {
+		workDir = c.replaceWithMountPoint(conf.WorkingDirectory)
+	}
+
+	absPath, commandLine, err := getApplicationName(commandLine, workDir, os.Getenv("PATH"))
```

Comment:
It's hard to follow what is happening with the commandLine here.  Why is assigned above and then re-assigned here?  Are they expected to be difference values?

Reply:
So the first thing we do is replace any instances of the env variable if there are any in the commandline. 

Then on line 184 we reassign commandLine as it may have changed depending on if it needed to be quoted or not. 

For instance, if someone passed `foo bar baz`, and foo bar.exe exists and baz was simply an argument, we need to quote the commandline to "foo bar" baz

Reply:
I'll comment the heck out of this, and re-push. It is a bit hard to follow

Reply:
I think  it is outside the scope of this PR but the `getApplicationName` isn't clear as to what it does and that caused some confusion when reading.  Comments help though it be better long term to refactor this a bit for clarity.  Maybe its harder than it looks... just an observation

Reply:
It's a pretty close interpretation of the c++ logic for windows server containers, so this likely played a role in its complexity :P. I'll keep a note of this.